includes/admin/OrderSettings.php

```diff
@@ -539,23 +539,23 @@ private function setLargeFormat(): void
      */
     private function setOnlyRecipient(): void
     {
-        $this->onlyRecipient = (bool) WCMP_Export::getChosenOrDefaultShipmentOption(
-            $this->shipmentOptions->hasOnlyRecipient(),
-            WCMYPA_Settings::SETTING_CARRIER_DEFAULT_EXPORT_ONLY_RECIPIENT,
-            $this->carrier
-        );
+        $settingName                      = WCMYPA_Settings::SETTING_CARRIER_DEFAULT_EXPORT_ONLY_RECIPIENT;
+        $onlyRecipientFromShipmentOptions = $this->shipmentOptions->hasOnlyRecipient();
+        $onlyRecipientFromSettings        = (bool)WCMYPA()->setting_collection->where('carrier', $this->carrier)->getByName($settingName);
+
+        $this->onlyRecipient = $onlyRecipientFromShipmentOptions ?: $onlyRecipientFromSettings;
```

Comment:
fromShipmentOptions zou null moeten kunnen zijn, alleen dan from settings? dus ?? ipv ?:, checken

Reply:
Ik heb dit nog even gecontroleerd maar ?? zal niet gaan werken omdat $signatureFromShipmentOptions geen null terug geeft maar false

Reply:
Maar even voor de helderheid, volgens de specs zou het wel null kunnen zijn (return value is ?bool), en in dat geval gaat dit stuk. Misschien kan het om een andere reden nooit gebeuren, en dan is dit natuurlijk goed.